### PR TITLE
adding dtype=str to all from_dict 

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Comment by antgonza** [comment]
@josenavas and @HannesHolste will review, thanks.

**Comment by josenavas** [comment]
Thanks @antgonza the changes here look good.

I'm assuming that the checks for making sure that everything is a string will come in a different PR? If so, I would also recommend removing the functions `type_loookup`, `get_datatypes`, `cast_to_python` and `as_python_types` from `qiita_db.metadata_template.util`; since those are the ones that are related with the type inference.

File: qiita_db/metadata_template/test/test_sample_template.py [hunk 1]
```diff
@@ -1342,20 +1343,22 @@ def test_update_numpy(self):
...
+        metadata = pd.DataFrame.from_dict(metadata_dict, orient='index',
+                                          dtype=str)
         npt.assert_warns(qdb.exceptions.QiitaDBWarning, st.update, metadata)
 
         sql = "SELECT * FROM qiita.sample_{0}".format(st.id)
         obs = self.conn_handler.execute_fetchall(sql)
-        exp = [['2.Sample1', False, datetime(2015, 9, 1)],
-               ['2.Sample2', False, datetime(2015, 9, 1)]]
+        exp = [['2.Sample1', 'false', '2015-09-01 00:00:00'],
+               ['2.Sample2', 'false', '2015-09-01 00:00:00']]
```
Comment (HannesHolste) [inline qiita_db/metadata_template/test/test_sample_template.py]: `2015-09-01 00:00:00` for sample2 -- should this correspond to the date for sample2 on line 1344 above?
```
'Sample2': {'bool_col': np.bool_(False),
                          'date_col': np.datetime64(datetime(2015, 8, 1))}
```
Current one is 9/1/15 and the other is 8/1/15

**Comment by HannesHolste** [comment]
Overall, looks good! 

(I assume you checked the locations in code beyond the scope of this that could be affected by changes in datatype – I'm still familiarizing myself with Qiita so I'm not sure where to look beyond the diffs here)

File: qiita_db/metadata_template/test/test_sample_template.py (hunk 1 above)
Comment (antgonza) [inline qiita_db/metadata_template/test/test_sample_template.py]: It changes bak to 09 in likne 1351.

**Comment by antgonza** [comment]
That's right, @josenavas! Thanks.

**Pull request merged**